src/eva/vision/utils/io/image.py

```diff
@@ -52,3 +53,40 @@ def read_image_as_array(path: str, flags: int = cv2.IMREAD_UNCHANGED) -> npt.NDA
         image = image[:, :, np.newaxis]
 
     return np.asarray(image).astype(np.uint8)
+
+
+def get_mask(
+    wsi: Any,
+    wsi_path: str,
+    level_idx: int,
+    kernel_size: tuple[int, int] = (7, 7),
+    gray_threshold: int = 220,
+    fill_holes: bool = False,
+) -> tuple[np.ndarray, float]:
+    """Extracts a binary mask from an image.
+    
+    Args:
+        image: The input image.
+        kernel_size: The size of the kernel for morphological operations.
+        gray_threshold: The threshold for the gray scale image.
+        fill_holes: Whether to fill holes in the mask.
+    """
+    image = np.array(
+        wsi.open_file(wsi_path).read_region(
```

Comment:
`open_file` here shouldn't be necessary here, `wsi` is already an initialized object (see `backends.wsi_backend(backend)(wsi_path)` in coordinates.py), so here, you can call `wsi.read_region(..)` instead

Reply:
Thanks, adjusting this. i also noticed in our implementation of `src/eva/vision/data/wsi/backends/openslide.py` we mixed up the order of the arguments (`location, size, level` vs `location, level, size` in openslide), changing it to `location, level, size` everywhere

Reply:
Good catch!